src/lib/mina_base/account.ml

```diff
@@ -819,35 +855,52 @@ let liquid_balance_at_slot ~global_slot (account : t) =
               ~vesting_period ~vesting_increment ~initial_minimum_balance ) )
       |> Option.value_exn
 
-let gen =
+let gen : t Quickcheck.Generator.t =
   let open Quickcheck.Let_syntax in
   let%bind public_key = Public_key.Compressed.gen in
   let%bind token_id = Token_id.gen in
   let%map balance = Currency.Balance.gen in
   create (Account_id.create public_key token_id) balance
 
-let gen_with_constrained_balance ~low ~high =
+let gen_with_constrained_balance ~low ~high : t Quickcheck.Generator.t =
   let open Quickcheck.Let_syntax in
   let%bind public_key = Public_key.Compressed.gen in
   let%bind token_id = Token_id.gen in
   let%map balance = Currency.Balance.gen_incl low high in
   create (Account_id.create public_key token_id) balance
 
-let gen_timing (account_balance : Balance.t) :
+let gen_any_vesting_range =
+  let open Quickcheck.Generator.Let_syntax in
+  let open Global_slot in
+  (* vesting period must be at least oe to avoid division by zero *)
+  let%bind vesting_period = Int.gen_incl 1 1000 >>| Global_slot.of_int in
+  let%bind vesting_end = gen_incl (of_int 1) max_value in
+  let%map cliff_time = gen_incl (of_int 1) vesting_end in
+  (cliff_time, vesting_end, vesting_period)
+
+let gen_at_least_one_vesting_period =
+  let open Quickcheck.Generator.Let_syntax in
+  let open Global_slot in
+  (* vesting period must be at least oe to avoid division by zero *)
+  let%bind vesting_period = Int.gen_incl 1 1000 >>| Global_slot.of_int in
```

Comment:
The fix here is that we need to be able to set the minimum vesting period here. We frequently do this with an optional parameter to the quickcheck generator function, such as `?(min_vesting_periods = 1)`. Then, from the test, call it like `gen_at_least_one_vesting_period ~min_vesting_periods:2`. Then you can actually remove the filter. I'm trying to avoid quickcheck generator filters in the codebase because they are a bad pattern that makes some tests run for way too long, since it just throws out values you generated rather than constraining the values you can generate.